Python/sort/merge_sort.py

```diff
@@ -0,0 +1,51 @@
+"""
+Merge Sort Algorithm Implementation in Python 3
+
+Purpose:
+It works on the principle of Divide and Conquer. Merge sort repeatedly breaks
+down a list into several sublists until each sublist consists of a single
+element and merging those sublists in a manner that results into a sorted list.
+
+Methodology:
+Given two sorted integer arrays A and B, merge B into A as one sorted array.
+
+Time Complexity:  O(n)
+Space Complexity: O(1)
+
+Note:
+You may assume that A has enough space (size that is greater or equal to m + n)
+to hold additional elements from B.
+The number of elements initialized in A and B are m and n respectively.
+
+"""
+
+
+class Solution:
+    """
+    @param A  a list of integers
+    @param m  an integer, length of A
+    @param B  a list of integers
+    @param n  an integer, length of B
+    @return nothing
+    """
+
+    def merge(self, A, m, B, n):
+        last, i, j = m + n - 1, m - 1, n - 1
+
+        while i >= 0 and j >= 0:
+            if A[i] > B[j]:
+                A[last] = A[i]
+                last, i = last - 1, i - 1
+            else:
+                A[last] = B[j]
+                last, j = last - 1, j - 1
+
+        while j >= 0:
+                A[last] = B[j]
+                last, j = last - 1, j - 1
+
+if __name__ == "__main__":
+    A = [1, 3, 5, 0, 0, 0, 0]
```

Comment:
@mehabhalodiya Take input from the user.

Reply:
Okay, I am removing class and then defining functions.
Further will take input from user.

Reply:
As with `class`, it will get more complicated.

Reply:
@mehabhalodiya Take dynamic input from user don't use static inputs